lib/hypervisor/hv_kvm/monitor.py

```diff
@@ -553,10 +553,8 @@ def HotDelDisk(self, devid):
     """
     self.execute_qmp("device_del", {"id": devid})
 
-    # TODO: implement receiving of QMP events
-    # We need to wait for the DEVICE_DELETED event via QMP before proceeding.
-    # The old implementation using HMP only worked because it is so slow.
-    time.sleep(1)
+    # wait for the DEVICE_DELETED event with one minute timeout
+    self.wait_for_qmp_event("DEVICE_DELETED", 1)
```

Comment:
The docstring of wait_for_qmp_event() says:

> If the timeout is reached, None returns.

Shouldn't we check for `None` value to see if the timeout has been reached and error out accordingly (e.g. throw `HypervisorError` )? Otherwise we would just wait one minute and then continue anyways (and also I assume the subsequent QMP call to `blockdev-del` would fail).

Reply:
You're right about the None test. I will add this.